Python/import.c

```diff
@@ -2438,6 +2443,28 @@ create_builtin(PyThreadState *tstate, PyObject *name, PyObject *spec)
     return mod;
 }
 
+PyObject*
+PyImport_CreateModuleFromInitfunc(
+    PyObject *spec, PyObject* (*initfunc)(void))
+{
+    PyThreadState *tstate = _PyThreadState_GET();
+
```

Comment:
I would suggest setting a ValueError if initfunc is `NULL`, or maybe a `SystemError` (call `_PyErr_BadInternalCall()`).